pumps/kafka.go

```diff
@@ -137,9 +142,21 @@ func (k *KafkaPump) Init(config interface{}) error {
 		k.log.WithField("SASL-Mechanism", k.kafkaConf.SASLMechanism).Warn("Tyk pump doesn't support this SASL mechanism.")
 	}
 
+	// Timeout is an interface type to allow both time.Duration and float values
+	var timeout time.Duration
+	switch k.kafkaConf.Timeout.(type) {
+	case string:
+		timeout, err = time.ParseDuration(k.kafkaConf.Timeout.(string))
+		if err != nil {
+			k.log.Fatal("Failed to parse timeout: ", err)
+		}
+	case float64:
+		timeout = time.Duration(k.kafkaConf.Timeout.(float64)) * time.Second
+	}
+
```

Comment:
if not string or float, it will default to `0s` is that expected?

Reply:
@sredxny it was the previous behavior before this change. Just in case, `0s` means no timeout from the Kafka library documentation